data/libs/EquipType.lua

```diff
@@ -325,17 +431,90 @@ function HyperdriveType:OnLeaveHyperspace(ship)
 	end
 end
 
+--==============================================================================
+
 -- NOTE: "sensors" have no general-purpose code associated with the equipment type
-local SensorType = utils.inherits(EquipType, "SensorType")
+---@class Equipment.SensorType : EquipType
+local SensorType = utils.inherits(EquipType, "Equipment.SensorType")
+
+--==============================================================================
 
 -- NOTE: all code related to managing a body scanner is implemented in the ScanManager component
-local BodyScannerType = utils.inherits(SensorType, "BodyScannerType")
+---@class Equipment.BodyScannerType : EquipType
+---@field stats table
+local BodyScannerType = utils.inherits(SensorType, "Equipment.BodyScannerType")
+
+--==============================================================================
+
+---@class Equipment.CabinType : EquipType
+---@field passengers Character[]?
+local CabinType = utils.inherits(EquipType, "Equipment.CabinType")
+
+---@param passenger Character
+function CabinType:AddPassenger(passenger)
+	table.insert(self.passengers, passenger)
```

Comment:
Q: No sanity-check required here that the cabin is already full?

Reply:
I'd prefer to check for this in `Passengers.EmbarkPassenger` instead. This is a "low level" method and cannot do anything useful other than throw a "stop the world" error if the cabin is full.

Reply:
.. which, IMHO, is still useful to catch programming errors.

Given we don't have proper debugging in Lua, I think it is valuable for low-level functions to have a lot of sanity checks.